jelares/portfolio/src/main/java/com/google/sps/servlet-data/PostBatch.java

```diff
@@ -0,0 +1,75 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.data;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/** Class containing a numPosts-large batch of posts */
+public final class PostBatch {
+  private final List<String> posts;
+  private int numPosts;
+
+  /**
+   * This ADT represents a mutable list of numPosts posts.
+   *
+   * Abstraction Function(posts) =  list of numPosts posts such that posts.get(i)
+   *                                represents post i where each post is a string of text.
+   * Representation Invariant: posts.size() == numPosts
+   * Saftey from Rep Exposure: all fields are never returned, numPosts only increased through
+   * mutator methods.
+   */
+
+  public PostBatch(List<String> posts) {
+    this.posts = new ArrayList<>(posts);
+    this.numPosts = posts.size();
+  }
+
+  /** Get the number of posts */
+  public int getNumPosts() {
+    return numPosts;
+  }
+
+  /** Get all posts as a string array */
+  public String[] getPostArray() {
```

Comment:
Is there a need to return String[] rather than List<String>? The latter return type would simplify implementation a lot, and also would likely be more useful for the consumer, since List has more API to offer.

```
public List<String> getPosts() {
  return posts;
}
```

Reply:
I made a new getPostList() method in case the client would like a list version like you describe. The reason an array version exists is to use with gson, because the only way I could see to easily encode the comments with gson was with an array of strings.

Reply:
I think json.toJson() also supports List<String>?